pkg/block/metrics.go

```diff
@@ -0,0 +1,133 @@
+package block
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/httputil"
+)
+
+type MetricsAdapter struct {
+	adapter Adapter
+}
+
+func NewMetricsAdapter(adapter Adapter) Adapter {
+	return &MetricsAdapter{adapter: adapter}
+}
+
+func (m *MetricsAdapter) InnerAdapter() Adapter {
+	return m.adapter
+}
+
+func (m *MetricsAdapter) Put(ctx context.Context, obj ObjectPointer, sizeBytes int64, reader io.Reader, opts PutOpts) error {
+	ctx = httputil.SetClientTrace(ctx, "block")
```

Comment:
I would prefer to keep the block adapter name somewhere on the client-trace.  I continue to believe that we will need to support multiple block adapters in future, and I am asking for very little  now: an additional parameter to the constructor, and a computed label such as "block-s3".
In any case, if you don't make this change then you should put `"block"` in some const.